code/game/machinery/camera/camera_assembly.dm

```diff
@@ -122,19 +124,18 @@
 
 	return ..()
 
-/obj/machinery/camera_assembly/proc/weld(obj/item/weapon/weldingtool/WT, mob/living/user)
-	if(busy)
-		return 0
+/obj/structure/camera_assembly/proc/weld(obj/item/weapon/weldingtool/WT, mob/living/user)
 	if(!WT.remove_fuel(0, user))
 		return 0
-
 	user << "<span class='notice'>You start to weld \the [src]...</span>"
 	playsound(src.loc, 'sound/items/Welder.ogg', 50, 1)
-	busy = 1

```

Comment:
busy stuff isn't necessary since the assembly is qdel()'d on success so do_after() fails to find the target in the other attempts.